src/shared/components/lazyMobXTable/LazyMobXTable.tsx

```diff
@@ -134,6 +134,35 @@ export function lazyMobXTableSort<T>(data:T[], metric:SortMetric<T>, ascending:b
     return dataAndValue.map(x=>x.data);
 }
 
+function getListOfEmptyStrings<T>(rowData:T) {
+    if (rowData instanceof Array) {
+        return Array(rowData.length).join(".").split(".");
+    } else {
+        return [""];
+    }
+}
+
+function getAsList<T>(rowData: T, data: string|string[]) {
+    let length = 1;
+    if (data instanceof Array) {
+        if (!(rowData instanceof Array) || rowData.length != data.length) {
+            throw new Error('Unexpected error!');
```

Comment:
This is not good because it makes assumptions about the form of rowData. For example, in some applications we don't need the output of download to be the same length as the input rowData.

Reply:
@adamabeshouse "unique per row" indeed does not necessarily need to be the length of rowData. But the assumption of "same order" will remain. This means that in the example where output of download is length 2 while rowData is 3 we would map the first and second rows to the first and second output of download data while mapping the third row of rowData to some empty string. 

Although I agree this is more flexible, I tend to prefer a more defensive implementation for such parts. I.e. with such a flexible implementation, my question then would be: which part of the code is responsible for checking the integrity of the data? I mean, how do you avoid the scenario where the 2 items of the output of download should actually be bound to rows 2 and 3 instead of 1 and 2? An implementation that is too flexible will just silently map things incorrectly.

Maybe it would be a good idea if you could share your feedback in the form of `code` for this part? Perhaps an alternative PR that achieves the same so we can more easily compare?

Reply:
If the user needs it to go to 2 and 3 then they can pad the beginning with an empty string. As I see it the issue you're bringing up is that if it's used improperly there won't be an easy way to see that, which is valid. In that case, we can, as you say, require that column.download(rowData) is the same length for all columns, without peeking into rowData itself at all.